src/nvim/api/command.c

```diff
@@ -791,29 +782,36 @@ static void build_cmdline_str(char **cmdlinep, exarg_T *eap, CmdParseInfo *cmdin
     kv_printf(cmdline, " %c", eap->regname);
   }
 
-  // Iterate through each argument and store the starting index and length of each argument
-  size_t *argidx = xcalloc(argc, sizeof(size_t));
   eap->argc = argc;
   eap->arglens = xcalloc(argc, sizeof(size_t));
+  size_t argstart_idx = cmdline.size;
   for (size_t i = 0; i < argc; i++) {
-    argidx[i] = cmdline.size + 1;  // add 1 to account for the space.
-    eap->arglens[i] = STRLEN(args[i]);
-    kv_printf(cmdline, " %s", args[i]);
+    String s = args.items[i].data.string;
+    eap->arglens[i] = s.size;
+    kv_concat(cmdline, " ");
+    kv_concat_len(cmdline, s.data, s.size);
+  }
+
+  // Done appending to cmdline, ensure it is NUL terminated
+  if (argc > 0) {
+    kv_push(cmdline, NUL);
```

Comment:
It might be because here null is not always written, while it should be done unconditionally?

Reply:
Already tried.